### PR TITLE
Added cfflib

### PR DESCRIPTION
**Pull request by umarbrowser**
HI, Please can you review and see if there was not any issue.

Thank you.

### CODE REVIEW

**Comment by umarbrowser** [comment]
Hi  @dougalsutherland can you please help me.

File: recipes/cfflib/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+  version: '{{ version }}'
+
+source:
+  fn: '{{ name }}-{{ version }}.{{ file_ext }}'
+  url: https://pypi.io/packages/source/{{ name[0] }}/{{ name }}/{{ name }}-{{ version }}.{{ file_ext }}
+  '{{ hash_type }}': '{{ hash_value }}'
+
+build:
+  number: 0
+  script: python -m pip install --no-deps --ignore-installed .
```
Comment (CJ-Wright) [inline recipes/cfflib/meta.yaml]: Could this be noarch?

**Comment by CJ-Wright** [comment]
This seems to depend on something called `loadsave` `ImportError: No module named 'loadsave`.

File: recipes/cfflib/meta.yaml (hunk 1 above)
Comment (umarbrowser) [reply recipes/cfflib/meta.yaml]: let me check ...

File: recipes/cfflib/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+
+source:
+  fn: '{{ name }}-{{ version }}.{{ file_ext }}'
+  url: https://pypi.io/packages/source/{{ name[0] }}/{{ name }}/{{ name }}-{{ version }}.{{ file_ext }}
+  '{{ hash_type }}': '{{ hash_value }}'
+
+build:
+  number: 0
+  noarch: python
+  script: 'python setup.py install '
```
[the remark this answers is not in the document]
Comment (umarbrowser) [reply recipes/cfflib/meta.yaml]: Ok, sir.

[the remark this answers is not in the document]
Comment (umarbrowser) [reply recipes/cfflib/meta.yaml]: Added it,

File: recipes/cfflib/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+  version: '{{ version }}'
+
+source:
+  fn: '{{ name }}-{{ version }}.{{ file_ext }}'
+  url: https://pypi.io/packages/source/{{ name[0] }}/{{ name }}/{{ name }}-{{ version }}.{{ file_ext }}
+  '{{ hash_type }}': '{{ hash_value }}'
+
+build:
+  number: 0
+  skip: true  # [py3k]
```
Comment (CJ-Wright) [inline recipes/cfflib/meta.yaml]: you might consider putting this as a python version constraint `python <3`

File: recipes/cfflib/meta.yaml [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+  script: python -m pip install --no-deps --ignore-installed .
+
+requirements:
+  build: 
+    - pip
+    - python
+    - numpy >= 1.2
```
Comment (CJ-Wright) [inline recipes/cfflib/meta.yaml]: Could this follow the numpy pinning guide? https://conda-forge.org/docs/meta.html#building-against-numpy

Comment (umarbrowser) [reply recipes/cfflib/meta.yaml]: Ok, thanks bro

File: recipes/cfflib/meta.yaml (hunk 3 above)
Comment (umarbrowser) [reply recipes/cfflib/meta.yaml]: Ok like this skip: true #[python <3] ??

Comment (CJ-Wright) [reply recipes/cfflib/meta.yaml]: more like this https://github.com/conda-forge/staged-recipes/pull/5654/files#diff-302a57ae4c9a65ce152a1f3d6a7fcf63R27 (you will be able to remove the skip entirely)

File: recipes/cfflib/meta.yaml [hunk 5]
```diff
@@ -15,19 +15,19 @@ source:
...
-  skip: true  # [py3k]
+  skip: true  # [python <3]
   script: python -m pip install --no-deps --ignore-installed .
 
 requirements:
   build: 
     - pip
     - python
-    - numpy >= 1.2
+    - numpy 1.2.*
```
Comment (CJ-Wright) [inline recipes/cfflib/meta.yaml]: Looking at the source code, does this require numpy to build?

File: recipes/cfflib/meta.yaml [hunk 6]
```diff
@@ -17,18 +17,15 @@ build:
...
-    - pip
-    - setuptools
     - python
-    - nibabel
     
   run: 
-    - python
-    - nibabel
-
+    - python # [py27]
```
Comment (CJ-Wright) [inline recipes/cfflib/meta.yaml]: I would handle this by python <3

**Comment by umarbrowser** [comment]
Oh, After conversation=="23", commit="31" i got it pass all the test.

thanks @CJ-Wright you help alot.

File: recipes/cfflib/meta.yaml [hunk 7]
```diff
@@ -0,0 +1,49 @@
...
+
+source:
+  fn: '{{ name }}-{{ version }}.{{ file_ext }}'
+  url: https://pypi.io/packages/source/{{ name[0] }}/{{ name }}/{{ name }}-{{ version }}.{{ file_ext }}
+  '{{ hash_type }}': '{{ hash_value }}'
+
+build:
+  number: 0
+  script: python -m pip install --no-deps --ignore-installed .
+  skip: True  # [py>27]
```
Comment (CJ-Wright) [inline recipes/cfflib/meta.yaml]: Could this be moved to a build/run constraint on python? `python <3`

**Comment by umarbrowser** [comment]
I am sorry i deleted them accidentally

File: recipes/cfflib/meta.yaml (hunk 7 above)
Comment (CJ-Wright) [reply recipes/cfflib/meta.yaml]: Right but that constraint would require python less than python 3 (python 2) to build/run the software.

Comment (CJ-Wright) [reply recipes/cfflib/meta.yaml]: You can remove the skip.

File: recipes/cfflib/meta.yaml (hunk 1 above)
Comment (CJ-Wright) [reply recipes/cfflib/meta.yaml]: Would it be possible to check this with noarch again?

**Comment by umarbrowser** [comment]
let me try.

**Comment by CJ-Wright** [comment]
@conda-forge/help-python could I get a second review on this? (there was a commit at one point that was deleting recipes and I just want to make sure this is safe to merge)

**Comment by umarbrowser** [comment]
Ok sir i am waiting :smile:

**Comment by umarbrowser** [comment]
Hello @CJ-Wright the @conda-forge/help-python link does not exists please can you review the link again.

Thankyou.

**Comment by CJ-Wright** [comment]
It exists but it is only accessible to members of conda-forge (It's an issue we are working on).
This LGTM so I'll merge.

**Comment by CJ-Wright** [comment]
Welcome to conda-forge!

**Comment by umarbrowser** [comment]
Thanks you sir.

**Pull request merged**